addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1214,6 +1213,16 @@ const UnitUserValueWidget = UserValueWidget.extend({
     _floatToStr: function (value) {
         return `${parseFloat(value.toFixed(5))}`;
     },
+    /**
+     * Checks whether the widget has a unit.
+     *
+     * @private
+     * @returns {Boolean} true if it has a unit, false otherwise.
+     */
+    _hasUnit() {
+        const params = this._methodsParams;
+        return !!params.unit;
```

Comment:
Let's not declare the `params` variable first, use `this._methodsParams` directly (also at other places).